service/storage_proxy.cc

```diff
@@ -2319,6 +2319,13 @@ future<> storage_proxy::send_hint_to_endpoint(frozen_mutation_and_schema fm_a_s,
             allow_hints::no);
 }
 
+future<> storage_proxy::mutate_hint_from_scratch(frozen_mutation_and_schema fm_a_s) {
+    // FIXME: using 1h as infinite timeout. If a node is down, we should get an
+    // unavailable exception.
+    const auto timeout = db::timeout_clock::now() + 1h;
```

Comment:
1 hour timeout again?

Reply:
I'm not sure I want to change the timeout in this patch. This is a backport, and the intent is to fix a data-corrupting behavior.

I agree that the timeout should be shorter, though, and I can see that I forgot to change it during review of https://github.com/scylladb/scylla/pull/5668 . How about a separate patch that can be backported after this?

Reply:
OK.

Reply:
That's the behavior before this PR. This change does not change the timeout length and since it's a bugfix for other problems we probably should not fix change timeouts here.